examples/gnrc_networking/tests/01-run.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+
+
+
+def _check_shell_is_working(child, tries=5):
```

Comment:
No need for loops, just do the following:
```suggestion
def _check_shell_is_working(child):
    """Just check that the shell is working.

    It ensures the application did not crash.
    """
    child.expect_exact('All up, running the shell now')
    child.sendline('help')
    child.expect_exact('Command              Description', timeout=1)
    child.expect_exact('---------------------------------------', timeout=1)
    child.expect_exact('> ', timeout=1)
```

Reply:
No it failed for some executions on my local board. It failed when running the `test_nordic_softdevice_openocd.sh`.

```
Welcome to pyterm!
Type '/exit' to exit.
2019-05-27 16:23:11,315 - INFO # main(): This is RIOT! (Version: buildtest)
2019-05-27 16:23:11,316 - INFO # RIOT network stack example application
2019-05-27 16:23:11,316 - INFO # All up, running the shell now
> help
2019-05-27 16:23:13,581 - INFO #  main(): This is RIOT! (Version: buildtest)
2019-05-27 16:23:13,584 - INFO # RIOT network stack example application
2019-05-27 16:23:13,587 - INFO # All up, running the shell now
> Timeout in expect script at "child.expect_exact('Command              Description', timeout=1)" (examples/gnrc_networking/tests/01-run.py:20)
```

I should also maybe increase the `timeout`.
I would split this in a dedicated pull request anyway.

Reply:
I just rechecked the proposed change using `test_nordic_softdevice_openocd.sh` and it works like a charm. Maybe you missed the synchronization `child.expect_exact('All up, running the shell now')`, at line 18 ?

Reply:
``` diff
diff --git a/examples/gnrc_networking/tests/01-run.py b/examples/gnrc_networking/tests/01-run.py
index bd778020d..baf937c7e 100755
--- a/examples/gnrc_networking/tests/01-run.py
+++ b/examples/gnrc_networking/tests/01-run.py
@@ -15,18 +15,12 @@ def _check_shell_is_working(child, tries=5):
 
     It ensures the application did not crash.
     """
-    for i in range(0, tries):
-        child.sendline('help')
-        try:
-            child.expect_exact('Command              Description', timeout=1)
-            child.expect_exact('---------------------------------------',
-                               timeout=1)
-            child.expect_exact('> ', timeout=1)
-            break
-        except Exception:
-            pass
-    else:
-        child.expect('>')
+    child.expect_exact('All up, running the shell now')
+    child.sendline('help')
+    child.expect_exact('Command              Description', timeout=1)
+    child.expect_exact('---------------------------------------',
+                        timeout=1)
+    child.expect_exact('> ', timeout=1)
 
 
 def testfunc(child):
```